metadata/metadata_tutorial_addition.Rmd

```diff
@@ -0,0 +1,353 @@
+
+# Set up
+
+## Load packages
+
+The following packages will need to be loaded: `RPostgres`, `DBI`, `dplyr`, `googlesheets`, `kableExtra`, and `glue`. The `metadata` upload package will first need to be installed from github and then loaded in. 
+
+```{r load-pack, message = FALSE}
+
+if(!require(metadata)){
+  install_github('kimberlyh66/betydbtools@metadata_package', subdir = 'metadata')
+}
+
+library(RPostgres)
+library(DBI)
+library(dplyr)
+library(googlesheets)
+library(kableExtra)
+library(glue)
+library(metadata)
+
+```
+
+## URL 
+
+Please provide the URL to your google sheet below. 
+
+A sample URL has been provided for a google sheet containing MAC season 8 metadata. Replace the sample URL with your own.
+
+```{r url}
+
+url <- 'https://docs.google.com/spreadsheets/d/1BPkZPfynZI9DB9bRJfa335uC_uNjST5jHVf-XUgSVvM/edit#gid=347205052' 
+
+```
+
+## Read in google sheets using URL
+
+Your metadata will be read in using the `googlesheets` package. Each sheet will be read in directly as a R object.
+
+```{r read-user-gs, message = FALSE}
+
+# get key from URL
+key <- extract_key_from_url(url)
+
+# register sheet 
+gs_obj <- key %>% gs_key(lookup = FALSE)
+
+# users sheet
+users <- gs_obj %>% gs_read(ws = 'users')
+
+# experiments sheet
+experiments <- gs_obj %>% gs_read(ws = 'experiments')
+
+# sites sheet
+sites <- gs_obj %>% gs_read(ws = 'sites')
+
+# treatments sheet
+treatments <- gs_obj %>% gs_read(ws = 'treatments')
+
+# cultivars sheet
+cultivars <- gs_obj %>% gs_read(ws = 'cultivars')
+
+# citations sheet
+citations <- gs_obj %>% gs_read(ws = 'citations')
+
+```
+
+# Upload data
+
+## Connect to database
+
+A connection to the test database needs to be created. Make sure that you have the test database running. 
+
+The port number should be the same as the one indicated in docker-compose.override.yml.
+
+```{r create-dbcon}
+dbcon <- dbConnect(RPostgres::Postgres(),
+                   host = 'localhost',
+                   user = 'bety',
+                   port = 5433)
+```
+
+## User id
+
+Your user id will be used when uploading your data. Your user id will be determined based on your provided user login (username for betydb account).
+
+```{r user-id}
+
+# get user id from user login
+bety_users <- tbl(dbcon, 'users') %>% collect()
+  
+user_id <- bety_users %>%
+  filter(login == users$login) %>%
+  select(id)
+
+```
+
+## Functions to be used for metadata upload
+
+You will be using the following functions from the `metadata` package to upload your data: 
+
+   * `add_experiments`
+   
+   * `add_sites`
+   
+   * `add_treatments`
+   
+   * `add_species`
+   
+   * `get_new_cultivars`
+   
+   * `add_cultivars`
+   
+   * `add_citations`
+
+## Step 1: Add new experiments
+
+To add new experiments, run `add_experiments`. The experiments google sheet object and your user id will need to be passed in.
+
+```{r add-experiments, results = 'hide'}
+
+add_experiments(dbcon, commit, user_id$id, experiments)
```

Comment:
All of the `add_*` lines are returning a bunch of TRUE statements, as shown below. I think this might have to do with the `commit` argument in the `add_row` function, but I wasn't able to figure out from briefly looking at the package code what is causing this. 

<img width="711" alt="Screen Shot 2019-05-20 at 8 37 52 AM" src="https://user-images.githubusercontent.com/6686508/58034087-9b641f00-7ada-11e9-8f98-ca49fc4b3605.png">

Reply:
I am not sure why there are no quotes around the date for your fetch experiments query. The query should look something like this according to the date: 

`select * from experiments where created_at::date = '2019-05-20'
and user_id = (select id from users where login = 'guestuser')`.

Did you run the following chunk? What value do you get for `today`?

```r
today <- Sys.Date()
```

Reply:
And I don't think we want it to return all these TRUEs? It's kind of distracting.

Reply:
I agree. I will hide these results in the chunk options